opm/parser/eclipse/EclipseState/Schedule/MSW/Segment.hpp

```diff
@@ -59,6 +59,11 @@ namespace Opm {
             return this->segmentType() == SegmentType::SICD;
         }
 
+        bool isValve() const
```

Comment:
Potentially, we will have many types here. Instead of adding a lot of isXXX function and if condition when using it, can we use switch to avoid adding these functions?

Reply:
It's a double-edged sword.  Writing this in terms of `switch` means making all calling code directly dependent on `Segment::SegmentType`.

Reply:
Okay, I could not see the problem with that. Do you mean when we update `Segment::SegmentType`, we might cause problem for the outputting code?

Reply:
It's about whether or not to make calling code dependent on the implementation details of class `Segment`.

Reply:
I've added a new commit that pulls these predicates out into free, inline functions in `Segment.hpp`.  I still think it's a good idea to have them because if you intentionally don't need/want to handle all possible enum values, then the `switch` statement will be needlessly verbose.

Reply:
Sure. It looks good.